pkg/receive/handler.go

```diff
@@ -339,6 +340,51 @@ func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 		tenant = h.options.DefaultTenantID
 	}
 
+	// Checking certs for possible value.
+	if h.options.TenantAttribute != "" {
+		if len(r.TLS.PeerCertificates) > 0 {
+			// First cert is the leaf authenticated against.
+			cert := r.TLS.PeerCertificates[0]
+
+			switch h.options.TenantAttribute {
+
+			case "organization":
+				if len(cert.Subject.Organization) > 0 {
+					tenant = cert.Subject.Organization[0]
+				} else {
+					http.Error(w, "could not get organization attribute from client cert", http.StatusBadRequest)
```

Comment:
All of these `else`s break the soft tenancy provided by allowing un-specified tenants to be ingested by a "default" hashring. This should be correctly handled to provide feature parity with the HTTP header approach IMO

Reply:
The whole idea is to limit tenant values to a trusted source and as such this needs to hard-fail. It's only so when enabling the feature, which I think is a reasonable thing to do. It ensures that no tenant could send data for a different tenant due to misconfiguration or intentional attack on a receive cluster. This can certainly be documented better which I will take care of. Same goes for it beating the header value in priority.